src/core/solver.py

```
@@ -108,67 +132,104 @@ class ScoringStrategy(ABC):
-            if weight > 0:
-                weight_m = max(1, round(weight * 100))
-            else:
-                weight_m = round(weight * 100)
-            vectors.append((col, scores, weight_m))
-        return vectors
+            raw_scores = [p.scores.get(col, 0.0) for p in participants]
+            scaled_raw = [round(s * config.SCALE_FACTOR) for s in raw_scores]
+            raw_total = sum(abs(s) for s in scaled_raw)
 
+            if raw_total == 0:
+                raise ValueError(f"Score dimension '{col}' has weight but sum is 0.")
```

Ruff (0.15.12)

[warning] Avoid specifying long messages outside the exception class

(TRY003)